versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/codecs/L2Codec.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.mongodb.codecs;
+
+import org.bson.BsonReader;
+import org.bson.BsonWriter;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import com.dremio.nessie.versioned.impl.L2;
+
+public class L2Codec implements Codec<L2> {
+  CodecRegistry codecRegistry;
+  public L2Codec(CodecRegistry codecRegistry) {
+    this.codecRegistry = codecRegistry;
+  }
+  @Override
+  public L2 decode(BsonReader bsonReader, DecoderContext decoderContext) {
+    bsonReader.readStartDocument();
+    //TODO complete this method.
+
```

Comment:
You should be using Entity for conversions. I suggest building a generic Entity <> Bson SerDe and then wrapping with class specific codecs.

Reply:
Yep. Agreed. Exactly the way I'm going. Next push will align more with that.

Reply:
We created a generic EntityCodec which is then instantiated in CodecProvider for each EntityType. EntityCodec then makes use of either BsonToEntityConverter or EntityToBsonConverter depending on direction which works at the Entity level, ie requires no knowledge of the specific codecs.